contrib/generate_kind_yaml.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# utility script to generate yaml config file with dynamic number of master 
+# and worker nodes to be used when creating a cluster. Rename the generated file to
+# kind.yaml or kind-ha.yaml as appropriate for the use case
+
+#PRE-REQUISITE: docker needs to be installed to run the docker image --> mikefarah/yq
+
+# Configurable definitons and defaults. The below can be set as environment variables as well
+MASTER_COUNT=${MASTER_COUNT:-1}
+NODE_COUNT=${NODE_COUNT:-2}
+MASTER_COUNT_MAX=${MASTER_COUNT_MAX:-3}
+NODE_COUNT_MAX=${NODE_COUNT_MAX:-15}
+OUTPUT_FILE=${OUTPUT_FILE:-gen_kind.yaml.GEN}
+
+echo INPUT::master_node_count=$MASTER_COUNT
+echo INPUT::worker_node_count=$NODE_COUNT
+echo MODULUS::mastercount%2 = `expr $MASTER_COUNT % 2`
+echo DEFAULTS::MASTER_COUNT_MAX=$MASTER_COUNT_MAX
+echo DEFAULTS::NODE_COUNT_MAX=$NODE_COUNT_MAX
+echo OUTPUT_FILE::Generated output filename=$OUTPUT_FILE
+
+# validate master node count
+if [[ $MASTER_COUNT -le 0 || $MASTER_COUNT -gt $MASTER_COUNT_MAX || `expr $MASTER_COUNT % 2` != 1 ]]
+then
+   echo "ERROR: master count valid value to be equal to between 1 and $MASTER_COUNT_MAX and should be ODD"
+   exit 1
+fi
+
+# validate worker node count
+if [[ $NODE_COUNT -lt 0 || $NODE_COUNT -gt $NODE_COUNT_MAX ]]
+then
+   echo "ERROR: worker count valid value to be between 0 and $NODE_COUNT_MAX"
+   exit 1
+fi
+
+TMPDIR=$(mktemp -d)
+COREFILE=kindGen.yaml
+MNODEFILE=masterGen.yaml
+WNODEFILE=nodeGen.yaml
+echo $TMPDIR $COREFILE $MNODEFILE $WNODEFILE
```

Comment:
Consider modifying this echo statement to reflect what is being printed. For example "Using the  temporary directory ...."
"Temporary file names are ..."